pkg/core/resources.go

```diff
@@ -25,10 +25,23 @@ type (
 
 	BaseConstructSet map[ResourceId]BaseConstruct
 
+	// Delete criteria is supposed to tell us when we are able to delete a resource based on its dependencies
+	DeleteCriteria struct {
+		// RequiresNoUpstream is a boolean that tells us if deletion relies on there being no upstream resources
+		RequiresNoUpstream bool
+		// RequiresNoDownstream is a boolean that tells us if deletion relies on there being no downstream resources
+		RequiresNoDownstream bool
+		// RequiresExplicitDelete is a boolean that tells us if deletion relies on the resource being explicitly deleted
+		RequiresExplicitDelete bool
+		// RequiresNoUpstreamOrDownstream is a boolean that tells us if deletion relies on there being no upstream or downstream resources
+		RequiresNoUpstreamOrDownstream bool
```

Comment:
Is this shorthand for a combination of `RequiresNoUpstream` and `RequiresNoDownstream`? If so, will it require us to make extra checks whenever we want to evaluate those constraints (potentially leading to bugs when we forget to check both formulations)?

Reply:
its the or case, so like for rds proxy target group or for a load balancer, we view those as connection pieces so if theres nothing up OR downstream then we can remove. For the and case we use the 2 above.

Also this is all from the spike so i assume we will model this better in the future as we learn